internal/abi/fastly/hostcalls_guest.go

```diff
@@ -98,24 +97,26 @@ func ParseUserAgent(userAgent string) (family, major, minor, patch string, err e
 		patchBuf  = prim.NewWriteBuffer(cap)
 	)
 
+	userAgentBuffer := prim.NewReadBufferFromString(userAgent).Wstring()
+
 	if err := fastlyUAPParse(
-		prim.NewReadBufferFromString(userAgent).Wstring(),
+		userAgentBuffer.Data, userAgentBuffer.Len,
 
-		familyBuf.Char8Pointer(),
+		uint32(uintptr(unsafe.Pointer(familyBuf.Char8Pointer()))),
```

Comment:
and similarly, a conversion function here that did all the necessary casting

Reply:
Nice idea.